module/VuFind/src/VuFind/ILS/Driver/SierraRest.php

```diff
@@ -2511,12 +2571,34 @@ function ($matches) {
      */
     protected function getBibRecord($id, $fields, $patron = false)
     {
-        return $this->makeRequest(
+        $cacheId = "bib|$id";
+        $fieldsArray = explode(',', $fields);
+        if ($cached = $this->getCachedData($cacheId)) {
+            if (!array_diff($fieldsArray, $cached['fields'])) {
+                // We already have all required fields cached:
+                return $cached['data'];
+            }
+        } else {
+            $cached = [
+                'fields' => [],
+                'data' => [],
+            ];
+        }
+        // Fetch requested fields as well as any cached fields to keep everything in
+        // sync:
+        $allFields = [...$fieldsArray, ...$cached['fields']];
```

Comment:
Will this automatically deduplicate, or should we do an array_unique?

Reply:
No, it's just like array_merge but faster (and doesn't handle string keys before PHP 8.2). array_unique added.